plugins/pluginmanagers/lazy.nix

```
@@ -1,222 +1,204 @@
-
-              module = helpers.mkNullOrOption (enum [ false ]) ''
-                Do not automatically load this Lua module when it's required somewhere
-              '';
-
-              priority = helpers.mkNullOrOption number ''
-                Only useful for start plugins (lazy=false) to force loading certain plugins first.
-                Default priority is 50. It's recommended to set this to a high number for colorschemes.
-              '';
-
-              optional = helpers.defaultNullOpts.mkBool false ''
-                When a spec is tagged optional, it will only be included in the final spec,
-                when the same plugin has been specified at least once somewhere else without optional.
-                This is mainly useful for Neovim distros, to allow setting options on plugins that may/may not be part
-                of the user's plugins
-              '';
-
-              opts =
-                with helpers.nixvimTypes;
-                helpers.mkNullOrOption (maybeRaw (attrsOf anything)) ''
-                  opts should be a table (will be merged with parent specs),
-                  return a table (replaces parent specs) or should change a table.
-                  The table will be passed to the Plugin.config() function.
-                  Setting this value will imply Plugin.config()
+                main = mkNullOrOption str ''
+                  You can specify the main module to use for config() and opts(),
+                  in case it can not be determined automatically. See config()
                 '';
-            };
-          });
-
-          listOfPlugins = types.listOf pluginType;
-        in
-        mkOption {
-          type = listOfPlugins;
-          default = [ ];
-          description = "List of plugins";
+
+                submodules = defaultNullOpts.mkBool true ''
+                  When false, git submodules will not be fetched.
+                  Defaults to true
+                '';
+
+                event =
+
+                  mkNullOrOption (maybeRaw (
+                    either str (listOf str)
+                  )) "Lazy-load on event. Events can be specified as BufEnter or with a pattern like BufEnter *.lua";
+
+                cmd = mkNullOrOption (maybeRaw (either str (listOf str))) "Lazy-load on command";
+
+                ft = mkNullOrOption (maybeRaw (either str (listOf str))) "Lazy-load on filetype";
+
+                keys = mkNullOrOption (maybeRaw (either str (listOf str))) "Lazy-load on key mapping";
+
+                module = mkNullOrOption (enum [ false ]) ''
+                  Do not automatically load this Lua module when it's required somewhere
+                '';
+
+                priority = mkNullOrOption number ''
+                  Only useful for start plugins (lazy=false) to force loading certain plugins first.
+                  Default priority is 50. It's recommended to set this to a high number for colorschemes.
```

```suggestion
                  Only useful for start plugins (lazy=false) to force loading certain plugins first.
                  
                  Default priority is 50. It's recommended to set this to a high number for colorschemes.
```
We should associate the lazy=false with the actual module option and maybe set the pluginDefault

---

I was trying to say I like it when our documentation references our actual module options verbiage that a user will be configuring the plugin with, like we do [here](https://github.com/nix-community/nixvim/blob/facf6b2d0c9e22d858956d1d458eac6baf155a08/plugins/cmp/default.nix#L20-L24). Instead of just saying `lazy=false` it's nice to point out what the actual config setting is the user should use, ie: `plugins.lazy.plugins.<plugin>.lazy = false`. Maybe it's too explicit, some usages we have just use the name of the setting itself. 

The default would be nice to be a part of the option definition, though. Instead of hiding it in the description, we can use `defaultNullOpts.mkInt 50 <description>` to set a pluginDefault that gets shown in the documentation.